src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -122,11 +122,28 @@ int BATT_SMBUS::task_spawn(int argc, char *argv[])
 		if (!is_running() && (busid == BATT_SMBUS_BUS_ALL || smbus_bus_options[i].busid == busid)) {
 
 			SMBus *interface = new SMBus(smbus_bus_options[i].busnum, BATT_SMBUS_ADDR);
+
+			if (interface == nullptr) {
+				PX4_WARN("Could not create SMBUS interface");
+				delete interface;
+				return PX4_ERROR;
+			}
+
+			//TODO: consider create the interface in the device
```

Comment:
There is an active todo here - is there a timeline for when this will be fixed? I'm not debating that this is a good incremental improvement we should get in, but I'm wondering if we might as well wrap it up.

Reply:
@LorenzMeier thank for the review.
At this point I wanted to fix more critical things I seen in the driver and not its structure.
anyway, this recent change (Drivers: SPI & I2C config simplifications (2) #14386). require me to re-base this pr. and this part is unnecessary. [here](https://github.com/PX4/Firmware/pull/14386/files#diff-70a4da136dd6e12ef0b4bef7e81eb5dbL86)
btw, does anyone know if locals changes I done. implementing the BQ40Z80, is relevant to the community?
changes has rebased and ready to re-review